client/state/shortcodes/reducer.js

```diff
@@ -17,6 +17,11 @@ import {
 	SHORTCODE_REQUEST_FAILURE,
 	SHORTCODE_REQUEST_SUCCESS,
 } from 'state/action-types';
+import { registerActionForward } from 'lib/redux-bridge';
+import { parse } from 'lib/shortcode';
+
+registerActionForward( 'RECEIVE_MEDIA_ITEMS' );
+registerActionForward( 'RECEIVE_MEDIA_ITEM' );
 
 const createRequestingReducer = requesting => {
```

Comment:
If the request was implemented with data layer, its deduplication feature could replace the `requesting` flag. Potential future improvement.

Reply:
totally